sklearn/semi_supervised/tests/test_label_propagation.py

```diff
@@ -13,67 +14,78 @@
 from numpy.testing import assert_array_almost_equal
 from numpy.testing import assert_array_equal
 
-ESTIMATORS = [
-    (label_propagation.LabelPropagation, {"kernel": "rbf"}),
-    (label_propagation.LabelPropagation, {"kernel": "knn", "n_neighbors": 2}),
-    (
-        label_propagation.LabelPropagation,
-        {"kernel": lambda x, y: rbf_kernel(x, y, gamma=20)},
-    ),
-    (label_propagation.LabelSpreading, {"kernel": "rbf"}),
-    (label_propagation.LabelSpreading, {"kernel": "knn", "n_neighbors": 2}),
-    (
-        label_propagation.LabelSpreading,
-        {"kernel": lambda x, y: rbf_kernel(x, y, gamma=20)},
-    ),
+
+ESTIMATOR_CLS = [label_propagation.LabelPropagation, label_propagation.LabelSpreading]
+
+
+PARAMETERS = [
+    {"kernel": "rbf"},
+    {"kernel": "knn", "n_neighbors": 2},
+    {"kernel": lambda x, y: rbf_kernel(x, y, gamma=20)},
 ]
 
 
-def test_fit_transduction():
-    samples = [[1.0, 0.0], [0.0, 2.0], [1.0, 3.0]]
-    labels = [0, 1, -1]
-    for estimator, parameters in ESTIMATORS:
-        clf = estimator(**parameters).fit(samples, labels)
-        assert clf.transduction_[2] == 1
+MATRIX_TYPES = (
+    sp.csc_matrix,
+    sp.csr_matrix,
+    sp.coo_matrix,
+    sp.dok_matrix,
+    sp.bsr_matrix,
+    sp.lil_matrix,
+    sp.dia_matrix,
+    np.array,
+)
 
 
-def test_distribution():
-    samples = [[1.0, 0.0], [0.0, 1.0], [1.0, 1.0]]
-    labels = [0, 1, -1]
-    for estimator, parameters in ESTIMATORS:
-        clf = estimator(**parameters).fit(samples, labels)
-        if parameters["kernel"] == "knn":
-            continue  # unstable test; changes in k-NN ordering break it
-            assert_array_almost_equal(
-                clf.predict_proba([[1.0, 0.0]]), np.array([[1.0, 0.0]]), 2
-            )
-        else:
-            assert_array_almost_equal(
-                np.asarray(clf.label_distributions_[2]), np.array([0.5, 0.5]), 2
-            )
-
-
-def test_predict():
-    samples = [[1.0, 0.0], [0.0, 2.0], [1.0, 3.0]]
+@pytest.mark.parametrize("matrix_type", MATRIX_TYPES)
+@pytest.mark.parametrize("estimator", ESTIMATOR_CLS)
+@pytest.mark.parametrize("parameters", PARAMETERS)
+def test_fit_transduction(matrix_type, estimator, parameters):
+    samples = matrix_type([[1.0, 0.0], [0.0, 2.0], [1.0, 3.0]])
```

Comment:
instead of `matrix_type`, we have a utility in testing to make the conversion. You can look at `sklearn.utils._testing._convert_container`. Like this, you can try array, and the two types of sparse matrix.